src/drivers/intel/ssp/ssp.c

```diff
@@ -273,7 +273,7 @@ static int ssp_set_config_tplg(struct dai *dai, struct ipc_config_dai *common_co
 	if (ssp->state[DAI_DIR_PLAYBACK] > COMP_STATE_READY ||
 	    ssp->state[DAI_DIR_CAPTURE] > COMP_STATE_READY) {
 		dai_info(dai, "ssp_set_config(): Already configured. Ignore config");
-		goto out;
+		goto clk;
```

Comment:
sorry, this doesn't look right to me.
When the SSP is already configured, you absolutely do not want to reconfigure it *again* and specifically touch the SSE bits.

Maybe I am missing something but the sequence and explanations are not quite right.

Reply:
@plbossart We will only config clock when config->flags is SOF_DAI_CONFIG_FLAGS_HW_PARAMS or SOF_DAI_CONFIG_FLAGS_HW_FREE. On the other hand driver will expect that clock will be configured when it set config->flags.
Currently, the first `ssp_set_config_tplg` is called by [sof_dai_config_setup](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/sof-audio.c#L47) and config->flags is SOF_DAI_CONFIG_FLAGS_NONE at that time. The second `ssp_set_config_tplg` is called by [hda_ctrl_dai_widget_setup](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L44) where we want to start the clock, but it will be ignored due to the SSP is already configured (but clock is not configured yet).

Reply:
ack, that makes sense now. I was only looking at the change of label and missed the additional switch. Thanks for the precisions @bardliao @brentlu